src/components/ButtonMenu.tsx

```diff
@@ -62,3 +74,10 @@ export type ImageMenuItemType = MenuItemBase & {
 export type MenuItem = MenuItemBase | IconMenuItemType | ImageMenuItemType;
 // This is done just to adapt to the React-Aria API for generating Sectioned lists of Menu Items.
 export type MenuSection = MenuItem & { items?: MenuItem[] };
+
+// Searching function the recuisively searches through the items through a built up regex and returns the items that match the search term. Maybe not necessary, as we could use .filter() instead.
+function match(term: string, items: MenuItem[]) {
+  const builtExp = Array.from(term).reduce((prevVal, curVal, i) => `${prevVal}[^${term.substr(i)}]*?${curVal}`, '');
+  const re = RegExp(builtExp);
+  return items.filter(({ label }) => label.toLowerCase().match(re));
+}
```

Comment:
**suggestion:** React-Aria has a `useFilter` hook, which has been helpful on the SelectField for searching through the items. Take a look at how that works here: https://github.com/homebound-team/beam/blob/main/src/inputs/internal/SelectFieldBase.tsx#L97